Exercise_1.java

```diff
@@ -11,19 +11,26 @@
   
     boolean isEmpty() 
     { 
-        if (top < 0) {
+        // To check if stack is empty, check if top element exists  
+        // if not, then stack is empty
+        if (top < 0) {   

```

Comment:
For reference purposes:
This if else block can be written in one line as well    return top < 0 ? true : false; 
or even better would be to write                                  return top == -1;     since top cannot go lower than -1